src/frontend/test_runner/tests/testdata/index.yaml

```diff
@@ -3,13 +3,13 @@
     create table t2 (v3 int, v4 numeric, v5 bigint);
     create index t1_v1 on t1(v1);
     create index t2_v3 on t2(v3);
-    /* should generate delta join plan, and stream index scan (not supported for now) */
+    /* should generate delta join plan, and stream index scan (pk not correct for now) */
     select * from t1, t2 where t1.v1 = t2.v3;
   stream_plan: |
-    StreamMaterialize { columns: [v1, v2, _row_id#0(hidden), v3, v4, v5, _row_id#1(hidden)], pk_columns: [_row_id#0, _row_id#1] }
-      StreamExchange { dist: HashShard([2, 6]) }
+    StreamMaterialize { columns: [v1, v2, _row_id#0(hidden), v3, v4, v5, _row_id#1(hidden)], pk_columns: [v1, _row_id#0, v3, _row_id#1] }
+      StreamExchange { dist: HashShard([0, 2, 3, 6]) }
         StreamDeltaJoin { type: Inner, predicate: $0 = $3 }
           StreamExchange { dist: HashShard([0]) }
-            StreamTableScan { table: t1, columns: [v1, v2, _row_id#0], pk_indices: [2] }
+            StreamIndexScan { index: t1_v1, columns: [v1, v2, _row_id#0], pk_indices: [0, 2] }
           StreamExchange { dist: HashShard([0]) }
-            StreamTableScan { table: t2, columns: [v3, v4, v5, _row_id#0], pk_indices: [3] }
+            StreamIndexScan { index: t2_v3, columns: [v3, v4, v5, _row_id#0], pk_indices: [0, 3] }
```

Comment:
An unrelated question: How to represent a 3-way delta join in plan?

Reply:
```
StreamDeltaJoin { type: Inner, join_keys: [[first-way keys], [second-way keys], [third-way keys]] }
  StreamIndexScan
  StreamIndexScan
  StreamIndexScan
```

Reply:
... we will only support joins like a.x = b.x = c.x; while a.x=b.x and a.y=c.y won't be supported.

Reply:
Let me think for a while if we can support things like `a.x=b.x and a.y=c.y`...

Reply:
Okay, I think if we want to support such joins, we cannot simply specify join keys in `StreamDeltaJoin`.

Firstly, frontend will need to select the indexes to use for this join. In this case, `a.x`, `a.y`, `b.x`, `c.y`. In this case, we are running a 3-way join, but we have 4 indexes.

So we can have 4 inputs for `StreamDeltaJoin`

```
StreamDeltaJoin { lookup_strategy }
  StreamIndexScan a.x
  StreamIndexScan a.y
  StreamIndexScan b.x
  StreamIndexScan c.y
```

And we create the 2*3 lookup strategy directly in optimizer. Meta can simply expand the lookup strategy to the expected stream fragments.

Reply:
there will be 3 elements in stream_input, and lookup_strategy will be a 2*3 matrix, where each element represents the arrangement side input of the node. If we want to construct the following fragments:

<img width="469" alt="image" src="https://user-images.githubusercontent.com/4198311/165044620-2e1320ee-ede6-4afb-a362-cd871c257042.png">

Then `stream_input = 0, 2, 3`, which represents the left-most 3 inputs. `lookup_strategy = [2, 3, 0, 3, 1, 2]`.

Reply:
cc @yuhao-su any ideas?

Reply:
It works, but has some slight drawbacks: the fragmenter has to add many `Exchange` to the plan, but this should be done by the optimizer.

Reply:
... and broadcast exchange if we want to re-use the `a.x` indexes in two lookups.

Reply:
> optimizer must guarantee that all stream index scans are of the same distribution of what delta join and lookup needs. So meta will only need to insert 1v1 exchange.

IIUC, there must be `Exchange` here, because the before lookup, the data must be distributed by join key. 

<img width="378" alt="Screen Shot 2022-04-25 at 4 04 27 PM" src="https://user-images.githubusercontent.com/10192522/165046516-c8152d3c-524a-47d3-b2a4-419d664f0653.png">

Reply:
After a second thought, that's true. There might be significant design changes in this case, lookup now needs to be shuffled 🤣 May need weeks to re-think about the current meta delta join rewriter. Currently it's too simple to handle this case.

Reply:
Previously I assumed all lookup and indexes will follow the same distribution. Then we have another reason to make `Lookup` part of the optimizer, instead of a pseudo delta join plan node.